beeai/supervisor/qe_data.py

```diff
@@ -0,0 +1,88 @@
+import asyncio
+from collections.abc import Callable
+from typing import Awaitable, Coroutine, TypeVar
+from pydantic import BaseModel
+
+from .http_utils import aiohttp_session
+
+QE_DATA_REPO = "https://gitlab.cee.redhat.com/otaylor/jotnar-qe-data"
+QE_DATA_URL = (
+    f"{QE_DATA_REPO}/-/raw/main/jotnar_qe_data.json?ref_type=heads&inline=false"
+)
+
+
+class TestLocationInfo(BaseModel):
+    component: str
+    qa_contact: str
+    tests_location: str | None = None
+    test_config_location: str | None = None
+    test_trigger_method: str | None = None
+    test_result_location: str | None = None
+    test_docs_url: str | None = None
+    notes: str | None = None
+
+
+R = TypeVar("R")
+
+
+def cache_async(
+    *,
+    max_age: float | None,
+) -> Callable[[Callable[[], Coroutine[None, None, R]]], Callable[[], Awaitable[R]]]:
```

Comment:
feels like this should be in `common/utils.py` or similar. it's very general purpose

Reply:
Fixed